autoscaler/controllers/common/processors.go

```diff
@@ -70,3 +78,76 @@ func GetGenericBatchProcessor() odigosv1.Processor {
 		},
 	}
 }
+
+func GenerateSourcesFilterProcessors(
+	ctx context.Context,
+	kubeClient client.Client,
+	dests *odigosv1.DestinationList,
+) (map[string]config.GenericMap, error) {
+	logger := log.FromContext(ctx)
+	sourcesFilterProcessors := make(map[string]config.GenericMap)
+
+	for _, dest := range dests.Items {
+
+		if dest.Spec.SourceSelector == nil {
+			continue
+		}
+
+		matchConditions := []string{}
+		if len(dest.Spec.SourceSelector.Namespaces) > 0 {
+			for _, namespace := range dest.Spec.SourceSelector.Namespaces {
+				namespaceSelectorKey := fmt.Sprintf("%s/*/*", namespace)
+				matchConditions = append(matchConditions, namespaceSelectorKey)
+			}
+		}
+		if len(dest.Spec.SourceSelector.Groups) > 0 {
+			matchedSources, err := fetchSourcesByGroups(ctx, kubeClient, dest.Spec.SourceSelector.Groups, logger)
+
+			if err != nil {
+				return nil, err
+			}
+
+			for _, source := range matchedSources {
+				sourceSelectorKey := fmt.Sprintf("%s/%s/%s", source.Spec.Workload.Namespace, source.Spec.Workload.Kind, source.Spec.Workload.Name)
```

Comment:
we already calculated this key in the `fetchSourcesByGroups`.

Can we reuse this key instead of calculating it again?